pkg/samples/list.go

```diff
@@ -24,172 +35,94 @@ func (sd *SampleData) GitRepo() string {
-	URL:         "https://github.com/stripe-samples/subscriptions-with-card-and-direct-debit",
-}
+func (s *Samples) getFromCacheOrGithub(noNetwork bool) error {
+	listPath, err := s.appCacheFolder("samples-list")
+	if err != nil {
+		return err
+	}
 
-var oxxoPayment = &SampleData{
-	Name:        "oxxo-payment",
-	Description: "Build a payment form to collect OXXO and card payments",
-	URL:         "https://github.com/stripe-samples/oxxo-payment",
-}
+	if _, err := s.Fs.Stat(listPath); os.IsNotExist(err) {
+		err = s.Git.Clone(listPath, "https://github.com/stripe-samples/samples-list.git")
+		if err != nil {
+			return err
+		}
+	} else if !noNetwork {
+		err := s.Git.Pull(listPath)
+		if err != nil {
+			if err != nil {
+				switch e := err.Error(); e {
+				case git.NoErrAlreadyUpToDate.Error():
+					// Repo is already up to date. This isn't a program
+					// error to continue as normal
+					break
+				default:
+					return err
+				}
+			}
+		}
+	}
 
-var auBecsPayment = &SampleData{
-	Name:        "au-becs-debit-payment",
-	Description: "Collecting AU BECS Direct Debit mandates and payments",
-	URL:         "https://github.com/stripe-samples/au-becs-debit-payment",
-}
+	file, err := afero.ReadFile(s.Fs, filepath.Join(listPath, "samples.json"))
+	if err != nil {
+		return err
+	}
 
-var sepaPayment = &SampleData{
-	Name:        "web-elements-sepa-debit-payment",
-	Description: "Accept SEPA Debit payments",
-	URL:         "https://github.com/stripe-samples/web-elements-sepa-debit-payment",
-}
+	var allSamples SampleList
 
-var idealPayment = &SampleData{
-	Name:        "web-elements-ideal-payment",
-	Description: "Build a payment form to collect iDEAL and card payments",
-	URL:         "https://github.com/stripe-samples/web-elements-ideal-payment",
-}
+	err = json.Unmarshal(file, &allSamples)
+	if err != nil {
+		return err
+	}
 
-var fpxPayment = &SampleData{
-	Name:        "web-elements-fpx-payment",
-	Description: "[Beta] Accept payments with FPX, a popular payment method in Malaysia",
-	URL:         "https://github.com/stripe-samples/web-elements-fpx-payment",
-}
+	for i, sample := range allSamples.Samples {
+		list[sample.Name] = &allSamples.Samples[i]
+	}
 
-var chargingSavedCard = &SampleData{
-	Name:        "charging-a-saved-card",
-	Description: "Learn how to charge a saved card",
-	URL:         "https://github.com/stripe-samples/charging-a-saved-card",
+	return nil
 }
 
-var connectOnboardingStandard = &SampleData{
-	Name:        "connect-onboarding-for-standard",
-	Description: "Learn how to onboard users for Stripe Connect with Onboarding for Standard",
-	URL:         "https://github.com/stripe-samples/connect-onboarding-for-standard",
-}
+// GetSamples returns a list that contains a mapping of Stripe Samples that
+// we want to be available in the CLI to some of their metadata.
+// TODO: what do we want to name these for it to be easier for users to select?
+// TODO: should we group them by products for easier exploring?
```

Comment:
@adreyfus-stripe do you have any thoughts here?

Reply:
being able to filter by product could be neat!

Reply:
I'll leave that for a separate sample. Discussing with @ctrudeau-stripe how to best approach this.